### PR TITLE
Export `LoadPjrtPlugin`, `InitializePjrtPlugin`, `GetCApiClient` functions

### PR DESCRIPTION
**Pull request by mofeing**
This functions allow to register external PjRt plugins (e.g. Metal or TPU).

### CODE REVIEW

File: deps/ReactantExtra/API.cpp [hunk 1]
```diff
@@ -143,23 +171,16 @@ extern "C" PjRtClient* MakeTPUClient(const char* tpu_path , const char** error)
...
-      *error = err;
-      return nullptr;
-    }
-    return xla::GetCApiClient("TPU").value().release();
+    const PJRT_Api* pluginLoad = LoadPjrtPlugin("tpu", tpu_library_path.c_str(), error);
+    if (pluginLoad == nullptr)
+        return nullptr;
+
+
+    auto tpu_status = InitializePjrtPlugin("tpu");
```
Comment (wsmoses) [inline deps/ReactantExtra/API.cpp:179]: this doesnt pass the error?

File: deps/ReactantExtra/API.cpp [hunk 2]
```diff
@@ -143,23 +171,16 @@ extern "C" PjRtClient* MakeTPUClient(const char* tpu_path , const char** error)
...
-    absl::Status tpu_status = pjrt::InitializePjrtPlugin("tpu");
-    if (!tpu_status.ok()) {
-      auto str = tpu_status.message();
-      char* err = (char*)malloc(str.size()+1);
-      memcpy(err, str.data(), str.size()+1);
-      *error = err;
-      return nullptr;
-    }
-    return xla::GetCApiClient("TPU").value().release();
+    const PJRT_Api* pluginLoad = LoadPjrtPlugin("tpu", tpu_library_path.c_str(), error);
```
Comment (wsmoses) [inline deps/ReactantExtra/API.cpp:174]: also unrelated but re above can you make it first check kEnvTpuLibraryPath then tpu_path ?

File: deps/ReactantExtra/BUILD [hunk 3]
```diff
@@ -272,6 +272,9 @@ cc_library(
 "-Wl,-exported_symbol,_enzymeActivityAttrGet",
 "-Wl,-exported_symbol,_MakeCPUClient",
 "-Wl,-exported_symbol,_MakeGPUClient",
+"-Wl,-exported_symbol,_LoadPjrtPlugin",
```
Comment (wsmoses) [inline deps/ReactantExtra/BUILD:275]: also add TPUClient here while at it?

File: deps/ReactantExtra/API.cpp (hunk 1 above)
Comment (mofeing) [reply deps/ReactantExtra/API.cpp]: ah right

File: deps/ReactantExtra/API.cpp (hunk 2 above)
Comment (mofeing) [reply deps/ReactantExtra/API.cpp:174]: sure

**Pull request merged**